Asynchronously load Jetpack header partner logos and social connect

sgomes opened the pull request:
All partner logos were being bundled together, when there will never be more than one (if any) shown. After this change, only the required logo will be loaded. In addition, the Jetpack header itself only needs to be shown on Jetpack sites, so after this change it too will be lazily loaded, conditionally.
A nice future improvement would be to move the SVG data out of JSX to static SVG files.

This PR also lazily loads the social connect components only when they're needed.

This PR appears to save ~36KB of compressed JS in total on `/log-in`.

#### Changes proposed in this Pull Request

* Lazily load partner logos with `AsyncLoad`.
* Lazily load `JetpackHeader` itself (only needed when site is Jetpack).
* Lazily load social connect component.

#### Testing instructions

* Test correct presence and behaviour of Jetpack header in Jetpack site login.
* Test correct behaviour of non-Jetpack site login.
* Test correct behaviour of social connect prompt.

sgomes committed:
Asynchronously load partner logos in Jetpack header.

All partner logos were being bundled together, when there will
never be more than one (if any) shown.

After this change, only the required logo will be loaded.

sgomes committed:
Asynchronously load Jetpack header.

Only load the `JetpackHeader` component in login if it will
actually be displayed.

sgomes committed:
Fix JetpackHeader tests after async load change.

sgomes committed:
Lazily load social login when needed.

blowery commented:
Probably need to rebase since https://github.com/Automattic/wp-calypso/pull/32099 landed

sgomes commented:
> Probably need to rebase since #32099 landed

I rebased, but there doesn't seem to have been any need for it. That PR only affected the example.

blowery commented:
> That PR only affected the example.

My bad, I got confused on what was changing.

sgomes commented:
> 
> My bad, I got confused on what was changing.

No worries, rebasing often is a good idea anyway :)

tyxla reviewed (changes requested):
Thanks for this! 👍 

Seems like the default AsyncLoad placeholder doesn't make much sense there:

![](https://cldup.com/8J_Vnb-9Yb.png)

We could fix that by hiding the placeholder with adding `placeholder={ null }` to the new `<AsyncLoad />` instances. I think it would be beneficial for all the instances touched in this PR.

sgomes committed:
Use full paths when requiring in AsyncLoad.

This helps provide more meaningful chunk names.

sgomes committed:
Remove placeholders on AsyncLoad.

sgomes commented:
> Seems like the default AsyncLoad placeholder doesn't make much sense there.
> 
> We could fix that by hiding the placeholder with adding `placeholder={ null }` to the new `<AsyncLoad />` instances. I think it would be beneficial for all the instances touched in this PR.

Great points, @tyxla, thanks! I added a null placeholder to all of the above.

On an unrelated note, I moved to using the full path for all of the `require`s as well, to improve the generated chunk names.

tyxla commented:
Thanks @sgomes! I'll take another look shortly.

tyxla reviewed (approved):
Looking and working well 👍 🚢 Thanks!
